### PR TITLE
add go example

### PR DESCRIPTION
**Pull request by sebastianwebber**
This PR adds an example to build go applications.

It adds:
 -  the input for gomod2nix
 - the output with the example app
 
 feel free to update anything needed or let me know what needs to be changed.

### CODE REVIEW

File: examples/go/.envrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source_url "https://raw.githubusercontent.com/cachix/devenv/95f329d49a8a5289d31e0982652f7058a189bfca/direnvrc" "sha256-d+8cBpDfDBj41inrADaJt+bDWhOktwslgoP5YiGJ1v0="
+
```
Comment (sebastianwebber) [inline examples/go/.envrc:2]: not sure if we need to remove this file

Comment (domenkozar) [reply examples/go/.envrc:2]: Yeah it's best to leave it out for examples, same for gitignore

**Comment by domenkozar** [comment]
Can you run pre-commit run -a

**Comment by sebastianwebber** [comment]
is this error expected?

```
✦2 ❯ devenv shell
• Building shell ...
• Using Cachix: devenv
warning: creating lock file '/Users/seba/projetos/github.com/sebastianwebber/devenv/examples/go/devenv.lock'
✔ Building shell in 6.7s.
• Entering shell
✨ devenv 1.1.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
(devenv) bash-5.2$ pre-commit -a
usage: pre-commit [-h] [-V] {autoupdate,clean,gc,init-templatedir,install,install-hooks,migrate-config,run,sample-config,try-repo,uninstall,validate-config,validate-manifest,help,hook-impl} ...
pre-commit: error: unrecognized arguments: -a
```

without the `-a`, it works:

```
(devenv) bash-5.2$ pre-commit
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /Users/seba/.cache/pre-commit/patch1726164631-49152.
golangci-lint........................................(no files to check)Skipped
gotest...............................................(no files to check)Skipped
govet................................................(no files to check)Skipped
[INFO] Restored changes from /Users/seba/.cache/pre-commit/patch1726164631-49152.
``` 

same for `devenv test`:

```shell
(devenv) bash-5.2$ devenv test
• Overriding .devenv to .devenv.NYFQiooY7g6D
• Building tests ...
• Using Cachix: devenv
warning: creating lock file '/Users/seba/projetos/github.com/sebastianwebber/devenv/examples/go/devenv.lock'
✔ Building tests in 7.9s.
warning: creating lock file '/Users/seba/projetos/github.com/sebastianwebber/devenv/examples/go/devenv.lock'
• Running tests ...
• Building shell ...
warning: creating lock file '/Users/seba/projetos/github.com/sebastianwebber/devenv/examples/go/devenv.lock'
✔ Building shell in 3.4s.
✨ devenv 1.1.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
• Setting up shell environment ...
✨ devenv 1.1.0 is newer than devenv input in devenv.lock. Run `devenv update` to sync.
• Testing ...
golangci-lint........................................(no files to check)Skipped
gotest...............................................(no files to check)Skipped
govet................................................(no files to check)Skipped
✔ Running tests in 6.1s.
• Tests passed :)
```

File: examples/go/.envrc (hunk 1 above)
Comment (sebastianwebber) [reply examples/go/.envrc]: updated commit without this files.

**Comment by domenkozar** [comment]
`(devenv) bash-5.2$ pre-commit -a`

You need to run `pre-commit run -a`.

**Review by 15159617213 — APPROVED** [review]

**Review by 15159617213 — APPROVED** [review]

**Comment by domenkozar** [comment]
@sebastianwebber did you run that in examples/go env?

**Comment by sebastianwebber** [comment]
hey @domenkozar , yes, i did:

```shell
❯ pwd
/Users/seba/projetos/github.com/sebastianwebber/devenv/examples/go

❯ devenv shell
• Building shell ...
• Using Cachix: devenv
warning: creating lock file '/Users/seba/projetos/github.com/sebastianwebber/devenv/examples/go/devenv.lock'
✔ Building shell in 68.9s.
• Entering shell
(devenv) bash-5.2$ pre-commit run -a
golangci-lint........................................(no files to check)Skipped
gotest...............................................(no files to check)Skipped
govet................................................(no files to check)Skipped
```

Should I run the command in another folder?

please let me know how to proceed.

**Pull request merged**